awx/main/utils/common.py

```diff
@@ -934,3 +935,17 @@ def has_model_field_prefetched(model_obj, field_name):
     # NOTE: Update this function if django internal implementation changes.
     return getattr(getattr(model_obj, field_name, None),
                    'prefetch_cache_name', '') in getattr(model_obj, '_prefetched_objects_cache', {})
+
+
+class cached_subclassproperty(object):
```

Comment:
Curious if we really need this for what we're using it for given my other comments.